server/src/main/java/io/crate/types/DoubleType.java

```diff
@@ -95,6 +96,17 @@ public Query rangeQuery(String field,
                 }
                 return null;
             }
+
+            @Override
+            public Query termsQuery(String field, List<Double> nonNullValues, boolean hasDocValues, boolean isIndexed) {
+                if (isIndexed) {
+                    return DoublePoint.newSetQuery(field, nonNullValues);
+                }
+                if (hasDocValues) {
+                    return SortedNumericDocValuesField.newSlowSetQuery(field, nonNullValues.stream().mapToLong(NumericUtils::doubleToSortableLong).toArray());
```

Comment:
values will be sorted already at the called `SortedNumericDocValuesSetQuery`
```suggestion
                    return SortedNumericDocValuesField.newSlowSetQuery(field, nonNullValues.stream().mapToLong(Double::longValue()).toArray());
```

Reply:
my understanding is that `NumericUtils::doubleToSortableLong` maps doubles to longs, `1.1` -> `4607182418800017408` such that it is sortable(which is what we need) while `Double::longValue()` drops decimals, `1.1` -> `1`, could you double check?

Reply:
Ah your completely right, ignore my comment.